autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -482,27 +578,57 @@ def __init__(
             metadata: dict | The metadata of the vector database. Default is None. If None, it will use this
                 setting: {"hnsw:space": "ip", "hnsw:construction_ef": 30, "hnsw:M": 16}. Creates Index on table
                 using hnsw (embedding vector_l2_ops) WITH (m = hnsw:M) ef_construction = "hnsw:construction_ef".
                 For more info: https://github.com/pgvector/pgvector?tab=readme-ov-file#hnsw
-            kwargs: dict | Additional keyword arguments.
+            model_name: str | Sentence embedding model to use. Models can be chosen from:
+                https://huggingface.co/models?library=sentence-transformers
 
         Returns:
             None
         """
-        if connection_string:
-            self.client = psycopg.connect(conninfo=connection_string, autocommit=True)
-        elif host and port and dbname:
-            self.client = psycopg.connect(
-                host=host, port=port, dbname=dbname, connect_timeout=connect_timeout, autocommit=True
+        try:
+            if connection_string:
+                parsed_connection = urllib.parse.urlparse(connection_string)
+                encoded_username = urllib.parse.quote(parsed_connection.username, safe="")
+                encoded_password = urllib.parse.quote(parsed_connection.password, safe="")
+                encoded_host = urllib.parse.quote(parsed_connection.hostname, safe="")
+                encoded_database = urllib.parse.quote(parsed_connection.path[1:], safe="")
+                connection_string_encoded = (
+                    f"{parsed_connection.scheme}://{encoded_username}:{encoded_password}"
+                    f"@{encoded_host}:{parsed_connection.port}/{encoded_database}"
+                )
+                self.client = psycopg.connect(conninfo=connection_string_encoded, autocommit=True)
+            elif host and port and dbname:
+                self.client = psycopg.connect(
+                    host=host,
+                    port=port,
+                    dbname=dbname,
+                    username=username,
+                    password=password,
+                    connect_timeout=connect_timeout,
+                    autocommit=True,
+                )
+        except psycopg.Error as e:
+            logger.error("Error connecting to the database: ", e)
+            raise e
+        self.model_name = model_name
+        try:
+            self.embedding_function = (
+                SentenceTransformer(self.model_name) if embedding_function is None else embedding_function
             )
-        self.embedding_function = (
-            SentenceTransformer("all-MiniLM-L6-v2") if embedding_function is None else embedding_function
-        )
+        except Exception as e:
+            logger.error(
+                f"Validate the model name entered: {self.model_name} "
+                f"from https://www.sbert.net/docs/pretrained_models.html\nError: {e}"
+            )
```

Comment:
please update the link. Thanks
```suggestion
                f"from https://huggingface.co/models?library=sentence-transformers\nError: {e}"
            )
```